components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/logout/idpinit/processor/FederatedIdpInitLogoutProcessor.java

```diff
@@ -0,0 +1,632 @@
+        }
+        return new LogoutResponse.LogoutResponseBuilder(HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+                OIDCAuthenticatorConstants.BackchannelLogout.LOGOUT_FAILURE_SERVER_ERROR);
+    }
+
+    private String getSessionIdFromSid(String sid) throws LogoutServerException {
+
+        try {
+            UserSessionDAO userSessionDAO = new UserSessionDAOImpl();
+            FederatedUserSession federatedUserSession = userSessionDAO.getFederatedAuthSessionDetails(sid);
+            if (federatedUserSession == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("No session information found for the sid: " + sid);
+                }
+                return null;
+            }
+            return federatedUserSession.getSessionId();
+        } catch (SessionManagementServerException e) {
+            throw handleLogoutServerException(ErrorMessages.RETRIEVING_SESSION_ID_MAPPING_FAILED, e, sid);
+        }
+    }
+
+    /**
+     * Terminate all the sessions of the user related sub claim.
+     *
+     * @throws LogoutServerException
+     * @throws SessionManagementException
+     */
+    private LogoutResponse.LogoutResponseBuilder logoutUsingSub(String userId)
+            throws LogoutServerException {
+
+        try {
+            if (StringUtils.isBlank(userId)) {
+                return new LogoutResponse.LogoutResponseBuilder(HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+                        ErrorMessages.LOGOUT_SERVER_EXCEPTION.getMessage());
+            }
+            UserSessionManagementService userSessionManagementService =
+                    OpenIDConnectAuthenticatorDataHolder.getInstance()
+                            .getUserSessionManagementService();
+            userSessionManagementService.terminateSessionsByUserId(userId);
+            if (log.isDebugEnabled()) {
+                log.debug("Sessions terminated for user Id: " + userId);
+            }
+            return new LogoutResponse.LogoutResponseBuilder(HttpServletResponse.SC_OK,
+                    OIDCAuthenticatorConstants.BackchannelLogout.LOGOUT_SUCCESS);
+        } catch (SessionManagementException e) {
+            throw handleLogoutServerException(ErrorMessages.USER_SESSION_TERMINATION_FAILURE, e, userId);
+        }
+    }
+
+    /**
+     * Validate the JWT token signature and the mandatory claim according to the OIDC specification.
+     *
+     * @param signedJWT
+     * @param identityProvider
+     * @return boolean value
+     * @throws LogoutClientException
+     * @throws LogoutServerException
+     */
+    private void validateLogoutToken(SignedJWT signedJWT, IdentityProvider identityProvider)
+            throws LogoutClientException, LogoutServerException {
+
+        try {
+            JWTClaimsSet claimsSet = signedJWT.getJWTClaimsSet();
+
+            if (!JWTSignatureValidationUtils.validateSignature(signedJWT, identityProvider)) {
```

Comment:
Can we have  JWTSignatureValidationUtils.validateSignature method to throw an exception when the validation fails?

Returning a boolean doesn't seem to add any value?